src/main/java/uk/gov/hmcts/reform/sendletter/services/LetterService.java

```diff
@@ -86,7 +86,9 @@ private void logMessageSendCompletion(Instant started, String messageId, Throwab
     private Message createQueueMessage(WithServiceName<Letter> letter, String messageId)
         throws JsonProcessingException {
 
-        insights.trackMessageReceived(letter.service, letter.obj.template, messageId);
+        letter.obj.documents
+            .forEach(document ->
+                insights.trackMessageReceived(letter.service, document.template, messageId));
```

Comment:
I Think there should be just one message sent to insights

Reply:
story said to record template as well

Reply:
We can record all templates in one entry.
It's one call to the service after all.

Plus we won't be able to distinguish requests to send multiple docs from the ones that send single doc if it stays like this.

Reply:
- request tracking filter will record each endpoint hit making it possible to do something like success/failure rate metrics, overall counters
- each message record will have all document metrics against which filters can be applied in azure app insights

moreover metric tracking has properties as `String -> String` map, would be inadequate to enlist in some way like `template1, template2..`

arguable point: include third parameter (metrics) in event tracking, saying which item in the list is it. worth adding